Model/Report.php

```diff
@@ -10,6 +10,9 @@
 
 class Report
 {
+    /** @var IP */
+    public $ip;
```

Comment:
Hmmm, what about adding a method (logBlockedRequest) and then $this->report->logBlockedRequest($rule) in the plugin? 

Allows us to kill the Logger dependency in plugin too.